pkg/kubelet/util/util_windows_test.go

```diff
@@ -31,17 +32,83 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetAddressAndDialer(t *testing.T) {
+
+	// Compare dialer function by pointer
+	tcpDialPointer := reflect.ValueOf(tcpDial).Pointer()
+	npipeDialPointer := reflect.ValueOf(npipeDial).Pointer()
+	var nilDialPointer uintptr = 0x0
+
+	tests := []struct {
+		endpoint      string
+		expectedAddr  string
+		expectedDial  uintptr
+		expectedError bool
+	}{
+		{
+			endpoint:      "tcp://localhost:15880",
+			expectedAddr:  "localhost:15880",
+			expectedDial:  tcpDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "npipe://./pipe/mypipe",
+			expectedAddr:  "//./pipe/mypipe",
+			expectedDial:  npipeDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "npipe:\\\\.\\pipe\\mypipe",
+			expectedAddr:  "//./pipe/mypipe",
+			expectedDial:  npipeDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "unix:///tmp/s1.sock",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+		{
+			endpoint:      "tcp1://abc",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+		{
+			endpoint:      "a b c",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		address, dialer, err := GetAddressAndDialer(test.endpoint)
+
+		dialerPointer := reflect.ValueOf(dialer).Pointer()
+		assert.Equal(t, test.expectedDial, dialerPointer)
+		assert.Equal(t, test.expectedAddr, address)
```

Comment:
To make easier to see what test case is failing when using table driven tests, you can use [subtests](https://blog.golang.org/subtests).

Reply:
I will try that!

Reply:
Hi! I followed your guide, but I don't know whether it is this. 😂